Tooltips

lindavin opened the pull request:
This pull request implements the tooltips on the home page. For some tooltips we use the TooltipHelpComponent. In other positions on the home page, it was easier to use an alternative implementation of tooltips: TooltipHelpComponentAlternative. 

This fixes #259. We will need to fully integrate this change by making tooltips for all pages in OED.

Note merging this changes the node modules.

lindavin committed:
Add most required tooltips on main page.

Add 8 of 10 tooltips on the main page because I could not figure out how to
insert the last two into the PlotlyChart.

lindavin committed:
Add some content for tooltips

lindavin committed:
Add styling to tooltip

lindavin committed:
Alternative implementation of tooltips

lindavin committed:
Replace STUBs with alternative tooltip component

lindavin committed:
Add the rest tooltip contents

lindavin committed:
Fix typos

lindavin committed:
Remove unnecessary data-html prop

lindavin committed:
Update react-tooltips

We first removed package-lock.json, and then modified the react-tooltips dependency
in package.json to version 4.2.10 and did npm install. This generates a
new package-lock.json; however, the depedency geojson-rewind was modified
unexpectedly. Additionally, this update to react-tooltips created errors in
the current usage of react-tooltips.

lindavin committed:
Fix import errors caused by upgrading to react-tooltips v4.2.10

lindavin committed:
Modify tooltiphelp to allow links

lindavin committed:
Add links object file

lindavin committed:
Modify components and structure of helpLinks.ts so that we can accomodate multiple links

lindavin committed:
Make tooltips clickable

lindavin committed:
Fix linting errors

sasankthapa commented:
I think everything is working as intended but there are some minor issues about styling like:

![image](https://user-images.githubusercontent.com/53589226/100422481-08516100-3050-11eb-8aa0-62e0360c27c2.png)

The buttons on the graph are messing with the texts and most of the links are wrapped outside the box.

huss commented:
While it would be nice to fix the formatting to avoid going outside the tooltip box, I suspect the box size is set and the link URL is too long. I think an easy fix that also makes sense is to use a name/words that link to the help page. The URL was put there to help plan the help page location but I don't see a lot of value in having it in the actual tooltip. Thus, I will edit the tooltip text to fix this and add it as needed.

As for the items on the page showing behind the tooltip and making it harder to read, I suspect this is designed to allow people to see the page while reading the tooltip. We could change (if that is an option) so the tooltip background is solid but I don't know if we want that all the time. I'm open to suggestions here.

huss commented on src/client/app/components/ChartLinkComponent.tsx:
```diff
@@ -5,6 +5,8 @@
 import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
 import { Button } from 'reactstrap';
+import TooltipHelpComponent from './TooltipHelpComponent';
```
This import does not seem to be used.

huss commented on src/client/app/components/DashboardComponent.tsx:
```diff
@@ -19,6 +19,8 @@ import { FormattedMessage } from 'react-intl';
 import { Dispatch, Thunk, ActionType } from '../types/redux/actions';
 
 import * as Plotly from 'plotly.js';
+import TooltipHelpComponent from './TooltipHelpComponent';
```
import does not seem to be used.

huss commented on src/client/app/components/HeaderButtonsComponent.tsx:
```diff
@@ -9,6 +9,9 @@ import { FormattedMessage } from 'react-intl';
 import MenuModalComponent from './MenuModalComponent';
 import { hasToken } from '../utils/token';
 import getPage from '../utils/getPage';
+import TooltipHelpComponent from './TooltipHelpComponent';
```
This and next imports do not seem to be used.

lindavin committed:
Merge branch 'development' into tooltips

lindavin committed:
Add rel attribute to resolve security warning

lindavin committed:
Replace text with link for formatting

huss committed:
simple help text changes - in progress

huss committed:
Merge branch 'tooltips' of https://github.com/lindavin/oed into pr/lindavin/506-1

lindavin committed:
Include links

lindavin committed:
Add constant

huss committed:
saving intermediate work

huss committed:
minor fixes

- Fixed double quote mechanism and
updated wording in help tips
- renamed old toottip so consistent with new scheme
- placed MPL in file
- Fixes a linting issue

huss committed:
forgotten file on last commit

lindavin committed:
Merge branch 'tooltips' of https://github.com/lindavin/OED into tooltips

lindavin committed:
Change explicit link to 'here'

lindavin committed:
Fix typo

lindavin committed:
Comment out all tooltips and add working solutions to tooltip and modal bug

lindavin committed:
Add the rest of the tooltips

lindavin committed:
Add missing tooltip

huss reviewed (changes requested):
The review chages are not showing up on GitHub so maybe I forgot to finish review. Doing it now.

lindavin committed:
Update tooltips so that only one is being used

lindavin committed:
Fix bug so tha the header tooltip works on every page

lindavin committed:
Add links for tooltips

lindavin committed:
Rebuild tooltips so that the export marker is detected

lindavin committed:
Add key to resolve react warning

lindavin committed:
Fix formatting issues

lindavin committed:
Update base link

huss commented:
This PR is now complete. It will likely be merged after the PRs that came before it (or sooner) and slated for version 0.7. I will do the final checks to be sure merging in the current development does not cause any issues.

I want to thank @lindavin for doing all of this and making changes as new ideas were thought of.

lindavin committed:
Rename TooltipMarker to TooltipMarkerComponent for consistency

lindavin committed:
Move bar stacking tip to following the current formatting for tips

lindavin committed:
Replace tooltip with alternative

This is an instance of adding the first tooltip to a page. One could look at this commit to see how to add tooltips to pages.
At this point there are no usages of TooltipHelpComponent.jsx; only TooltipHelpComponentAlternative.jsx is used.

lindavin committed:
Rename tooltip imports

huss commented:
I have reviewed the recent changes (that I asked for). All still seems fine. Note I may be changing the links in the tooltips as I create the help pages and desire changes.

huss committed:
new links for help web pages

- The new help web pages (currently only on @huss fork) changed links
and these are now updated for tooltips.
- I also added a commented out line that allows the web address to be
someone's own GitHub web page if they have a fork and set it up
to serve that page. Used for debugging.

huss reviewed (approved):
I have looked these over and all is fine. Changes requested plus more done.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/lindavin/506-1
